src/test/java/org/opensearch/ad/ml/CheckpointDaoTests.java

```diff
@@ -1098,4 +1100,138 @@ public void testShouldSave() {
         // 1658863778000L + 6 hrs < Instant.now
         assertTrue(checkpointDao.shouldSave(Instant.ofEpochMilli(1658863778000L), false, Duration.ofHours(6), clock));
     }
+
+    // This test is intended to check if given a checkpoint created by RCF-3.0-rc3 ("rcf_3_0_rc3_single_stream.json")
+    // and given the same sample data will rc3 and current RCF version (this test originally created when 3.0-rc3 is in use)
+    // will produce the same anomaly scores.
+    // The scores in this method were produced from AD running with RCF3.0-rc3 dependency
+    // and this test runs with the most recent RCF dependency that is being pulled by this project.
+    public void testDeserialize_rcf3_rc3_single_stream_model() throws Exception {
+        // Model in file rc1_trcf_model_direct is a checkpoint creatd by RCF-3.0-rc1
+        URI uri = ClassLoader.getSystemResource("org/opensearch/ad/ml/rcf_3_0_rc3_single_stream.json").toURI();
+        String filePath = Paths.get(uri).toString();
+        String json = Files.readString(Paths.get(filePath), Charset.defaultCharset());
+        // For the parsing of .toTrcf to work I had to manually change "\u003d" in code back to =.
+        // In the byte array it doesn't seem like this is an issue but whenever reading the byte array response into a file it
+        // converts "=" to "\u003d" https://groups.google.com/g/google-gson/c/JDHUo9DWyyM?pli=1
+        // I also needed to bypass the trcf as it wasn't being read as a key value but instead part of the string
+        Map map = gson.fromJson(json, Map.class);
+        String model = (String) ((Map) ((Map) ((ArrayList) ((Map) map.get("hits")).get("hits")).get(0)).get("_source")).get("modelV2");
+        // model = model.split(":")[1].substring(1);
```

Comment:
nit: previously i think I added this to deal with "trcf" was this not needed any more in this case?